### PR TITLE
fix: select menu options to accept both rest and array

### PR DESCRIPTION
**Pull request by imranbarbhuiya**
**Please describe the changes this PR makes and why it should be merged:**
followup of  #7874 
**Status and versioning classification:**

### CODE REVIEW

File: packages/discord.js/src/structures/SelectMenuBuilder.js [hunk 1]
```diff
@@ -23,12 +23,13 @@ class SelectMenuBuilder extends BuildersSelectMenu {
 
   /**
    * Adds options to this select menu
-   * @param {APISelectMenuOption[]} options The options to add to this select menu
+   * @param {RestOrArray<APISelectMenuOption>} options The options to add to this select menu
    * @returns {SelectMenuBuilder}
    */
-  addOptions(options) {
+  addOptions(...options) {
+    const _options = normalizeArray(options);
```
Comment (SpaceEEC) [inline packages/discord.js/src/structures/SelectMenuBuilder.js:30]:
```suggestion
```

File: packages/discord.js/src/structures/SelectMenuBuilder.js [hunk 2]
```diff
@@ -23,12 +23,13 @@ class SelectMenuBuilder extends BuildersSelectMenu {
...
     return super.addOptions(
-      options.map(({ emoji, ...option }) => ({
+      _options.map(({ emoji, ...option }) => ({
```
Comment (SpaceEEC) [inline packages/discord.js/src/structures/SelectMenuBuilder.js:32]:
```suggestion
      normalizeArray(options).map(({ emoji, ...option }) => ({
```

File: packages/discord.js/src/structures/SelectMenuBuilder.js [hunk 3]
```diff
@@ -37,12 +38,13 @@ class SelectMenuBuilder extends BuildersSelectMenu {
 
   /**
    * Sets the options on this select menu
-   * @param {APISelectMenuOption[]} options The options to set on this select menu
+   * @param {RestOrArray<APISelectMenuOption>} options The options to set on this select menu
    * @returns {SelectMenuBuilder}
    */
-  setOptions(options) {
+  setOptions(...options) {
+    const _options = normalizeArray(options);
```
Comment (SpaceEEC) [inline packages/discord.js/src/structures/SelectMenuBuilder.js:45]:
```suggestion
```

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
The original PR inlined the normalization where possible or shadowed options where it wasn't.
I think it'd be good to keep it like that:

(I can't merge those suggestions because there is a deleted line there :( )

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**